geom.go

```diff
@@ -182,6 +182,20 @@ func NewRect(p Point, lengths []float64) (r *Rect, err error) {
 	return
 }
 
+// NewRectFromPoints constructs and returns a pointer to a Rect given a corner points.
+func NewRectFromPoints(minPoint, maxPoint Point) (r *Rect, err error) {
+	if len(minPoint) != len(maxPoint) {
+		err = &DimError{len(minPoint), len(maxPoint)}
+		return
+	}
```

Comment:
can you add a case to swap `minPoint` and `maxPoint` in case they're provided in the wrong order, and a test case to check this?